File: src/main/java/io/georocket/index/Indexer.java
package io.georocket.index;

import java.util.Map;

import io.georocket.util.XMLStreamEvent;

/**
 * Indexes chunks
 * @author Michel Kraemer
 */
public interface Indexer {
  /**
   * Will be called on every XML event in the chunk
   * @param event the event
   */
  void onEvent(XMLStreamEvent event);
  
  /**
   * Will be called when the whole chunk has been passed to the indexer
   * @return the results that should be added to the index or an empty
   * map if nothing should be added (never <code>null</code>)
   */
  Map<String, Object> getResult();
}

File: src/main/kotlin/io/georocket/index/Indexer.kt
package io.georocket.index

import io.georocket.util.StreamEvent

/**
 * Indexes chunks
 * @author Michel Kraemer
 */
interface Indexer<T : StreamEvent> {
  /**
   * Will be called on every stream [event] in the chunk
   */
  fun onEvent(event: T)

  /**
   * Will be called when the whole chunk has been passed to the indexer.
   * Returns the results that should be added to the index or an empty
   * map if nothing should be added.
   */
  fun makeResult(): Map<String, Any>
}
